File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/visitor/hosting/ContainerService.java
package com.bigboxcode.designpattern.visitor.hosting;

public class ContainerService implements Service {

    private final double price = 5.60;
    private int quantity;

    public ContainerService(int quantity) {
        this.quantity = quantity;
    }

    public double getPrice() {
        return price;
    }

    public int getQuantity() {
        return quantity;
    }

    @Override
    public double accept(HostingCalculatorVisitor hostingCalculatorVisitor) {
        return hostingCalculatorVisitor.visit(this);
    }
}

File: design-patterns/php/visitor/hostingcost/ContainerService.php
<?php
// ContainerService.php

namespace BigBoxCode\DesignPattern\Visitor\HostingCost;

class ContainerService implements Service {
    private const PRICE = 5.60;
    private int $quantity;

    public function __construct(int $quantity) {
        $this->quantity = $quantity;
    }

    public function getPrice(): float {
        return self::PRICE;
    }

    public function getQuantity(): int {
        return $this->quantity;
    }

    public function accept(HostingCalculatorVisitor $hostingCalculatorVisitor): float {
        return $hostingCalculatorVisitor->visitContainerService($this);
    }
}